crates/vault-registry/src/types.rs

```diff
@@ -108,6 +116,91 @@ pub type DefaultVaultId<T> = VaultId<<T as frame_system::Config>::AccountId, Cur
 
 pub type DefaultVaultCurrencyPair<T> = VaultCurrencyPair<CurrencyId<T>>;
 
+pub mod upgrade_vault_release {
+
+    use super::*;
+    use frame_support::weights::Weight;
+
+    /// If a pending client release exists, set the current release to that.
+    /// The pending release becomes `None`.
+    pub fn try_upgrade_current_vault_release<T: Config>() -> Weight {
+        let writes: Weight = if let Some(pending_release) = Pallet::<T>::pending_client_release() {
+            log::info!("Upgrading current vault release.");
+            crate::CurrentClientRelease::<T>::put(pending_release.clone());
+            let no_release: Option<ClientRelease<T::Hash>> = None;
+            crate::PendingClientRelease::<T>::put(no_release);
+            Pallet::<T>::deposit_event(crate::Event::<T>::ApplyClientRelease {
+                release: pending_release,
+            });
+            2
+        } else {
+            log::info!("No pending vault release, skipping migration.");
+            0
+        };
```

Comment:
```suggestion
        let writes: Weight = if let Some(pending_release) = PendingClientRelease::<T>::take() {
            log::info!("Upgrading current vault release.");
            crate::CurrentClientRelease::<T>::put(pending_release.clone());
            Pallet::<T>::deposit_event(crate::Event::<T>::ApplyClientRelease {
                release: pending_release,
            });
            2
        } else {
            log::info!("No pending vault release, skipping migration.");
            0
        };
```

Reply:
i.e. `PendingClientRelease::take()` and then there is no need to overwrite storage

Reply:
Nice, thanks for explaining